taichi/rhi/metal/metal_device.mm

```diff
@@ -179,9 +179,13 @@
   return *this;
 }
 
-MetalCommandList::MetalCommandList(const MetalDevice &device)
-    : device_(&device) {}
-MetalCommandList::~MetalCommandList() {}
+MetalCommandList::MetalCommandList(const MetalDevice &device,
+                                   MTLCommandQueue_id cmd_queue)
+    : device_(&device) {
+  cmdbuf_ = [[cmd_queue commandBuffer] retain];
```

Comment:
This should be guarded by `@autoreleasepool` I suppose?

Reply:
I guess if I do `[cmd_queue commandBuffer]` it returns a object with references already incremented once? If that's the case and if we are releasing manually in dtor I don't think this should be in autorelease?